src/ir/tag_check.h

```diff
@@ -37,6 +37,11 @@ class TagCheck {
         "\").\n");
   }
 
+  bool operator==(const TagCheck &other) const {
```

Comment:
Why weren't these needed earlier?

Reply:
We're not currently putting in any effort to force these things to be unique. Stating an equivalent `TagCheck` multiple times in the Datalog is idempotent, similar with `Claim`, `Edge`, etc. As such, we are just putting our "sets" of facts into `std::vector`s, which don't require hash and `==`. In addition, I think the parser is probably preventing cases of non-uniqueness from coming up. This might be useful in production code later, though.